### PR TITLE
[wallet] Use client-api-schema

### PR DESCRIPTION
**Pull request by snario**
Uses `client-api-schema` in the `wallet`.

### CODE REVIEW

File: packages/wallet/src/utils/json-rpc-utils.ts [hunk 1]
```diff
@@ -3,68 +3,36 @@ import {
...
-}
-
-function createAllocationOutcomeFromParams(params: JsonRpcAllocations): Outcome {
+import {
+  UpdateChannelParams,
+  Allocations,
+  CreateChannelParams
+} from "@statechannels/client-api-schema";
+
+// TODO: Figure out how these should have been used
```
Comment (snario) [inline packages/wallet/src/utils/json-rpc-utils.ts]: @lalexgap

Comment (lalexgap) [reply packages/wallet/src/utils/json-rpc-utils.ts]: These should probably be moved somewhere else. These are interfaces for messages between `wallet`s.

The json-rpc api specifies that the `data` on a message can be anything. These interfaces define what the `wallet` currently expects/handles in the `data` field.

```
{
  "jsonrpc": "2.0",
  "method": "PushMessage",
  "id": 1,
  "params": {
    "recipient": "user123",
    "sender": "user456",
    "data": "0x123.."
  }
}
```

File: packages/wallet/src/redux/sagas/messaging/message-handler.ts [hunk 2]
```diff
@@ -108,11 +113,12 @@ function* handleJoinChannelMessage(payload: RequestObject) {
 function* handlePushMessage(payload: RequestObject) {
   // TODO: We need to handle the case where we receive an invalid wallet message
   const {id} = payload;
-  const message = payload.params as JsonRpcMessage;
+  const message = payload.params as PushMessageParams;
   if (isRelayableAction(message.data)) {
     yield put(message.data);
     yield fork(messageSender, outgoingMessageActions.postMessageResponse({id}));
   } else {
+    // TODO: Add schema for PushMessageParams.data to client-api-schema/schema/push-message.json
```
Comment (lalexgap) [inline packages/wallet/src/redux/sagas/messaging/message-handler.ts]: I don't think we want to specify the `data` in the JSON schema. The specific messaging format shouldn't really be part of the API that the JSON schema specifies. The client should just blindly pass along `data`.

Comment (snario) [reply packages/wallet/src/redux/sagas/messaging/message-handler.ts]: Oh okay, I'll remove the todo.
```suggestion
```

**Pull request merged**